eth/vm/stack.py

```diff
@@ -22,69 +23,213 @@
+        the stack.
+
+        Items are ordered with the top of the stack as the first item in the tuple.
+
+        Note: This function is optimized for speed over readability.
+        """
+        if num_items > len(self.values):
+            raise InsufficientStack(
+                "Wanted %d stack items, only had %d",
+                num_items,
+                len(self.values),
+            )
+        else:
+            neg_num_items = -1 * num_items
+
+            all_popped = reversed(self.values[neg_num_items:])
+            del self.values[neg_num_items:]
+
+            type_cast_popped = []
+
+            for item_type, popped in all_popped:
+                if item_type is int:
+                    type_cast_popped.append(popped)
+                elif item_type is bytes:
+                    type_cast_popped.append(big_endian_to_int(popped))  # type: ignore
                 else:
-                    yield big_endian_to_int(value)
-            elif type_hint == constants.BYTES:
-                value = self.values.pop()
-                if isinstance(value, bytes):
-                    yield value
+                    raise _busted_type(item_type, popped)
+
+            return tuple(type_cast_popped)  # type: ignore
+
+    def pop_bytes(self, num_items: int) -> Tuple[bytes, ...]:
+        """
+        Pop and return a tuple of bytes of length ``num_items`` from the stack.
+
+        Raise `eth.exceptions.InsufficientStack` if there are not enough items on
+        the stack.
+
+        Items are ordered with the top of the stack as the first item in the tuple.
+
+        Note: This function is optimized for speed over readability.
+        """
+        if num_items > len(self.values):
+            raise InsufficientStack(
+                "Wanted %d stack items, only had %d",
+                num_items,
+                len(self.values),
+            )
+        else:
+            neg_num_items = -1 * num_items
+
+            all_popped = reversed(self.values[neg_num_items:])
+            del self.values[neg_num_items:]
+
+            type_cast_popped = []
+
+            for item_type, popped in all_popped:
+                if item_type is int:
+                    type_cast_popped.append(int_to_big_endian(popped))  # type: ignore
+                elif item_type is bytes:
+                    type_cast_popped.append(popped)  # type: ignore
                 else:
-                    yield int_to_big_endian(value)
-            elif type_hint == constants.ANY:
-                yield self.values.pop()
-            else:
-                raise TypeError(
-                    "Unknown type_hint: {0}.  Must be one of {1}".format(
-                        type_hint,
-                        ", ".join((constants.UINT256, constants.BYTES)),
-                    )
-                )
+                    raise _busted_type(item_type, popped)
+
+            return tuple(type_cast_popped)
```

Comment:
Is this actually faster than `to_tuple` and yield?

Reply:
Yup